Sources/PubNubChat/Storage/CoreData/CoreDataMigrationManager.swift

```diff
@@ -63,20 +64,45 @@ class DefaultCoreDataMigrationManager: CoreDataMigrationManager {
       return
     }
     
-    for modelVersion in nextModelVersions {
+    let allVersionFiles = modelBundle.paths(
+      forResourcesOfType: "mom",
+      inDirectory: modelURL.lastPathComponent
+    )
+    
+    let sortedManagedObjectModels = allVersionFiles.compactMap() {
+      NSManagedObjectModel(contentsOf: URL(fileURLWithPath: $0))
+    }.sorted() {
+      $0.integerValue < $1.integerValue
+    }
+    
+    let currentStoreMetadata = try NSPersistentStoreCoordinator.metadataForPersistentStore(
+      ofType: NSSQLiteStoreType,
+      at: persistentStoreLocation
+    )
+    
+    let identifiers = (currentStoreMetadata["NSStoreModelVersionIdentifiers"] as? [String] ?? []).first ?? String()
+    let integerVersionFromTheStore = Int(identifiers) ?? 0
+    let modelVersionMatchingStore = sortedManagedObjectModels.first(where: { $0.integerValue == integerVersionFromTheStore })!
```

Comment:
I think it should throw instead of crash, or it should be protected earlier

Reply:
Right, this is what should happen. I will double-check it and let you know.

Reply:
I changed it slightly so now the `migrateIfNeeded` method returns a boolean value. I checked this and it will dump the old data if it fails.